### PR TITLE
[FIX] l10n_ar: Responsibility type "IVA No Alcanzado"

### PR DESCRIPTION
**Pull request by bruno-zanotti**
**Description of the issue/feature this PR addresses:**

- Add a new responsibility type "IVA No Alcanzado" and update the partner "AFIP" with that responsibility.

- Fix a bug when a partner has a not handled responsibility type, returning an empty list as possibles document types.

**Current behavior before PR:**
1. The "AFIP" partner has a wrong responsibility type.
2. If a partner has a responsibility type that we are not handling at the time of returning the allowed letters it will show a traceback error.

**Desired behavior after PR is merged:**
1. The "AFIP" partner has the correct responsibility type (IVA No Alcanzado).
2. If a partner has a responsibility type that we are not handling at the time of returning the allowed letters it will return an empty list so the user will not be able to select any document type.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by vbe-odoo** [comment]
Move to technical for review @jco-odoo 
thanks
cc @ren-odoo @Bruno-Zanotti

File: addons/l10n_ar/models/account_journal.py [hunk 1]
```diff
@@ -56,23 +57,19 @@ def _get_journal_letter(self, counterpart_partner=False):
...
-        if not counterpart_partner:
-            return letters
-
-        if not counterpart_partner.l10n_ar_afip_responsibility_type_id:
-            letters = []
-        else:
-            counterpart_letters = letters_data['issued' if self.type == 'purchase' else 'received'][
-                counterpart_partner.l10n_ar_afip_responsibility_type_id.code]
+        letters = letters_data['issued' if self.type == 'sale' else 'received'].get(
+            self.company_id.l10n_ar_afip_responsibility_type_id.code, [])
```
Comment (jco-odoo) [inline addons/l10n_ar/models/account_journal.py:69]: .get seems unnecessary

**Review by jco-odoo — CHANGES_REQUESTED** [review]

File: addons/l10n_ar/models/account_journal.py (hunk 1 above)
Comment (jco-odoo) [reply addons/l10n_ar/models/account_journal.py:69]: The user should not change AFIP responsibilities...  And I don't see why we should not provide all the combinations then.

Comment (Bruno-Zanotti) [reply addons/l10n_ar/models/account_journal.py:69]: Okay, maybe we should put a warning if the user tries to change the responsibility of the company.. and about the combinations, the remaining ones are responsibilities no much used that are not implemented yet. I will change this, but maybe we can talk about these points better in the meeting.

Comment (Bruno-Zanotti) [reply addons/l10n_ar/models/account_journal.py]: @jco-odoo Done!

**Comment by jco-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**